pkg/reconciler/chart/factory.go

```diff
@@ -326,3 +360,106 @@ func (f *DefaultFactory) Delete(version string) error {
 	}
 	return err
 }
+
+func (f *DefaultFactory) componentBaseDir(c *Component) string {
+	return filepath.Join(f.storageDir, gitComponentsBaseDir,
+		fmt.Sprintf("%x-%s", sha1.Sum([]byte(c.url)), c.name)) //nolint
+}
+
+func (f *DefaultFactory) indicatorExistsOrClean(baseDir string) (bool, error) {
+	if file.DirExists(baseDir) {
+		if file.Exists(filepath.Join(baseDir, wsReadyIndicatorFile)) {
+			// if the component repo is fully cloned just fetch to update it.
+			return true, nil
+		}
+		// broken clone, clean it
+		f.logger.Warnf("Deleting workspace '%s' because previous download/clone does not contain all the required files", baseDir)
+		if err := os.RemoveAll(baseDir); err != nil {
+			return false, err
+		}
+
+	}
+	return false, nil
+}
+
+func (f *DefaultFactory) fetchComponent(component *Component, dstDir string) error {
+	dstPath := path.Join(dstDir, component.name)
+	f.logger.Infof("Fetching GIT repository '%s' in workspace '%s'",
+		component.url, dstPath)
+
+	repo := &reconciler.Repository{
+		URL: component.url,
+	}
+
+	tokenNamespace := component.configuration["repo.token.namespace"]
+	if tokenNamespace != nil {
+		repo.TokenNamespace = fmt.Sprintf("%s", tokenNamespace)
+	}
+	clientSet, err := reconcilerK8s.NewInClusterClientSet(f.logger)
+	if err != nil {
+		return err
+	}
+	cloner, _ := git.NewCloner(&git.Client{}, repo, true, clientSet, f.logger)
+
+	return cloner.FetchAndCheckout(dstPath, component.version)
+}
+
+// getLatestRevOfVersion works on local cache. If "version" is a branch, it will retrun the revisionID of it's HEAD
+// if "version" is a tag, it will return it's revisionID.
+func (f *DefaultFactory) getLatestRevOfVersion(version, path string) (string, error) {
```

Comment:
Unfortunately no. This function uses a "local" git client since it only needs to open the local cache. Both `clone` and `fetch` need to have access to the upstream with a remote client and auth. I believe it would be better to keep the logic here separated.